format/spec.md

```diff
@@ -309,10 +309,10 @@ Partition specs capture the transform from table data to partition values. This
 | **`identity`**    | Source value, unmodified                                     | Any                                                                                                       | Source type |
 | **`bucket[N]`**   | Hash of value, mod `N` (see below)                           | `int`, `long`, `decimal`, `date`, `time`, `timestamp`, `timestamptz`, `string`, `uuid`, `fixed`, `binary` | `int`       |
 | **`truncate[W]`** | Value truncated to width `W` (see below)                     | `int`, `long`, `decimal`, `string`                                                                        | Source type |
-| **`year`**        | Extract a date or timestamp year, as years from 1970         | `date`, `timestamp`, `timestamptz`                                                                        | `int`       |
-| **`month`**       | Extract a date or timestamp month, as months from 1970-01-01 | `date`, `timestamp`, `timestamptz`                                                                        | `int`       |
-| **`day`**         | Extract a date or timestamp day, as days from 1970-01-01     | `date`, `timestamp`, `timestamptz`                                                                        | `int`      |
-| **`hour`**        | Extract a timestamp hour, as hours from 1970-01-01 00:00:00  | `timestamp`, `timestamptz`                                                                                        | `int`       |
+| **`years`**       | Extract a date or timestamp year, as years from 1970         | `date`, `timestamp`, `timestamptz`                                                                        | `int`       |
```

Comment:
This might impact Trino, Dremio and other engines which has naming conventions according to spec. 

https://trino.io/docs/current/connector/iceberg.html#partitioned-tables
https://docs.dremio.com/cloud/reference/sql/commands/create-table/

IMO, better to change the spark transform name and deprecate the old ones.

Reply:
cc: @aokolnychyi, @RussellSpitzer

Reply:
I can do that.  

Would it be useful to keep the unintended "backward" compatibility though?  Or do you think just force the Spark API to match the spec?

Reply:
We need to maintain backward compatibility. I would suggest adding the singular options and also keeping the plural ones as well. They don't conflict and avoids people to run into errors. Also curious about what others think of it.

Reply:
I added singular options to each Spark3 version.  How does this look?

I can add some tests as well if you think that would be useful.

Reply:
> I can add some tests as well if you think that would be useful.

You can add a test in `TestAlterTablePartitionFields` to make sure that we don't break this in the future. It could be something similar as `testSparkTableAddDropPartitions` where you use the singular transform names.

> Also, should I change the prefix of this PR to Spark: ?

Yes please :)

Reply:
For the tests, what if I replicate [these](https://github.com/clettieri/iceberg/blob/docs-fix-partition-transforms/spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAlterTablePartitionFields.java#L164) but for some or all of `year, month, hour, day` ?

Reply:
Added the tests. 

I ran `/gradlew :iceberg-spark:iceberg-spark-extensions-3.4_2.12:test --tests TestAlterTablePartitionFields` locally and saw things pass.  LMK if I can do anything else here (I'll try to catch up on the contributing guidelines tomorrow).  Thanks!